packages/@aws-cdk/aws-lambda/lib/function-base.ts

```diff
@@ -182,7 +182,7 @@ export abstract class FunctionBase extends Resource implements IFunction {
   /**
    * Whether the addPermission() call adds any permissions
    *
-   * True for new Lambdas, false for imported Lambdas (they might live in different accounts).
+   * True for new Lambdas, false for version $LATEST.
```

Comment:
Will this now work for imported lambdas in other accounts? 

If not, maybe we should check the account id from the ARN in the `fromXXX()` API

Reply:
Does this mean we would force users to set up a `env` for the account id? Because if not the account id will only get resolved at deploy time so we wouldn't be able to check if the arns matched. Unless there is another way to get the account id that I do not know of...

Reply:
For imported lambda functions, account id should be available in the ARN specified during import.

Reply:
yes but in order to check the account id with the users account, we would need to get the account tied to the stack right? but if the `env` (using a Context Provider) is not supplied to the app, then the token will only be resolved at deploy time.

Reply:
Fair enough. 

You could add a check if the user had in fact called it using [`withResolved()`](https://github.com/aws/aws-cdk/blob/1fe9684ea6b2dcaac1d97b64edfd4ef87cc65c0f/packages/%40aws-cdk/core/lib/token.ts#L199)